packages/grafana-ui/src/types/datasource.ts

```diff
@@ -199,6 +201,10 @@ export abstract class DataSourceApi<
     options?: TContextQueryOptions
   ) => Promise<DataQueryResponse>;
 
+  getTagKeys?(options: any): any[];
+
+  getTagValues?(options: { key: any }): any[];
```

Comment:
should be Promise<MetricNameMeta[]> 

which is a new type, not sure what the name should, just a {text: string, value: string} 
same type that metricFindQuery and getTagKeys returns

Reply:
Added `metricFindQuery` to `DatasourceAPI` and now uses a `Promise<MetricFindValue>` as returned type. `getTagKeys` and `getTagValues` uses the same returned type. Hard with naming, but I think `MetricFindValue` could work in the first iteration.